src/main/scala/org/ergoplatform/nodeView/state/DigestState.scala

```diff
@@ -126,9 +125,21 @@ class DigestState protected(override val version: VersionTag,
 
   def close(): Unit = store.close()
 
-  private def update(newVersion: VersionTag, newRootHash: ADDigest): Try[DigestState] = Try {
+  private def update(header: Header): Try[DigestState] = {
+    val version: VersionTag = VersionTag @@ header.id
+    val newContext = stateContext.appendHeader(header)
+    val cb = ByteArrayWrapper(ErgoStateReader.ContextKey) -> ByteArrayWrapper(newContext.bytes)
```

Comment:
Why to store the state context on disk on every header, if we can trivially get it from history during startup (e.g. in ErgoNodeViewHolder), and then hold it in memory only?

Reply:
Unfortunately we can not keep them just in memory, because in addition to startup, we should also recover it on rollback, and `rollbackTo` method only contains `version`, we can not recover height (and other possibly required fields) if we do not keep them in the same DB we rollback.